plugins/BotTracking/RecordBuilders/AIAssistantReports.php

```diff
@@ -216,4 +205,49 @@ private function populateTableForActionType(array $tables, int $actionType, LogA
             ]);
         }
     }
+
+    private function queryBotRequests(LogAggregator $logAggregator, int $actionType)
+    {
+        $where  = $logAggregator->getWhereStatement('bot', 'server_time');
+
+        $sql = sprintf(
+            "SELECT * FROM (SELECT bot.bot_name, log_action.name AS url, COUNT(*) AS requests
+             FROM %s AS bot
+             INNER JOIN %s AS log_action ON log_action.idaction = bot.idaction_url
+             WHERE log_action.name IS NOT NULL
+               AND log_action.name <> ''
+               AND log_action.type = %d
+               AND %s
+             GROUP BY bot.bot_name, url WITH ROLLUP) AS rollupQuery
+             ORDER BY bot_name, requests DESC, url",
```

Comment:
```suggestion
             ORDER BY requests DESC, bot_name, url",
```

Otherwise it could happen that the ranking limit will hit and we have only one bot in the final report. Instead of the (what I would expect) top actions across all bots,

Should be verifiable by adding a second page hit for `Perplexity-User`.

In `RankingQueryApiTest::testRankingQueryUsesOthersRowPages` it would be archived as `| Perplexity-User | __mtm_ranking_query_others__ |        2 |`. But with two page views on the same action it should be the top entry and not be displayed as "Others".

Reply:
Good catch 👍